Merge pre master commits

SheerSt opened the pull request.

AwfullyWaffley commented:
Yeah, in principle I'm not sure why it would be that much slower, but it must have something to do with the check it does to see if it's on the correct thread - maybe it takes longer if it isn't? I didn't really time that specific part.

SheerSt commented:
@AwfullyWaffley  I was actually able to keep all of your changes. I didn't think of this yesterday but there should be a way to skirt around the save file issue by just forcing the new isEgg format to adhere to the same format as previous. Let me know if this looks okay to you and I will merge it.

After that I will need to push my recent changes on master, resolve any conflicts between the load-time-reduction and master, and then merge it into master.

AwfullyWaffley reviewed (approved):
Looks good to me!

kferraga committed:
Merge pull request #5 from SheerSt/main

Unifying side branch

kferraga committed:
Adding finished sprites

Should be all Pokemon pictures from 8/20 to 9/14 in finished sprites

Squashed to adjust changes
